pkg/kubelet/network/cni/cni.go

```diff
@@ -129,24 +138,38 @@ func (plugin *cniNetworkPlugin) TearDownPod(namespace string, name string, id ku
 // TODO: Use the addToNetwork function to obtain the IP of the Pod. That will assume idempotent ADD call to the plugin.
 // Also fix the runtime's call to Status function to be done only in the case that the IP is lost, no need to do periodic calls
 func (plugin *cniNetworkPlugin) GetPodNetworkStatus(namespace string, name string, id kubecontainer.ContainerID) (*network.PodNetworkStatus, error) {
-	netns, err := plugin.host.GetRuntime().GetNetNS(id)
+	plugin.mu.Lock()
+	defer plugin.mu.Unlock()
+
+	netnsPath, err := plugin.host.GetRuntime().GetNetNS(id)
 	if err != nil {
 		return nil, err
 	}
 
-	nsenterPath, lookupErr := exec.LookPath("nsenter")
-	if lookupErr != nil {
-		return nil, fmt.Errorf("Unable to obtain IP address of container: missing nsenter.")
+	nsenterPath, err := plugin.getNsenterPath()
+	if err != nil {
+		return nil, err
 	}
-
-	ip, err := network.GetPodIP(utilexec.New(), id, nsenterPath, netns, network.DefaultInterfaceName)
+	// Try to retrieve ip inside container network namespace.
+	ip, err := network.GetPodIP(plugin.execer, id, nsenterPath, netnsPath, network.DefaultInterfaceName)
 	if err != nil {
 		return nil, err
 	}
 
 	return &network.PodNetworkStatus{IP: ip}, nil
 }
 
+func (plugin *cniNetworkPlugin) getNsenterPath() (string, error) {
+	if plugin.nsenterPath == "" {
```

Comment:
Could we read the nsenterPath in Init() maybe?  Then it's always available and we could just use plugin.nsenterPath instead of calling the function.